samples/csharp/getting-started/DatabaseIntegration/README.md

```diff
@@ -0,0 +1,28 @@
+# Using a database as a data source
+This sample demonstrates how to use a database as a data source for an ML.Net pipeline using the IDataView interface. The data can then be used for training and testing a model. In order
+to support mulitple databass the Entity Framework is used as part of this sample.
+
+## Problem
+
+In scenarios where the dataset lives in a database, the dataset is likely large and associated but not live within a single table. In addition, having to massage the data to be consumeable into an machine learning pipeline can be an expensive and tedious chore.
+
+
+## Solution
+This sample shows how to use the Entity Framework to connect to a database and leverage the existing schema allowing to query for the data that is needed to train and make this available to ML.Net through the IDataView interface.
+
+This sample uses SQLite to help demonstrate the database integration, but any database that is supported by the Entity Framwork can be used. This sample also uses ML.Net's ability to convert an IEnumerable to and IDataView -- as the Entity Framework will expose the results of a query as an IEnumberable. To get started the following commands need to run to setup the database. To create the database, the dot net migrations commands are used.
```

Comment:
I would then mention that we used a no tracking query to not require all data from being loaded in memory at the same time here.

Reply:
Diego - I found this URL that talks about NoTracking, but it discusses about how its great for read-only scenarios in that it does not have to track changes on each item. However, I don't see any mention about it not requiring all data being loaded into memory at the same time. Do you have more references to this that I can refer to in the example?
[https://docs.microsoft.com/en-us/ef/core/querying/tracking](https://docs.microsoft.com/en-us/ef/core/querying/tracking)


---
In reply to: [270257293](https://github.com/dotnet/machinelearning-samples/pull/295#discussion_r270257293) [](ancestors = 270257293)

Reply:
I was purposedly providing you with a bit more information than what we currently have in the documentation :smile: 

The truth is that no tracking queries in EF Core 1.x and 2.x are not always streaming and the rules are tricky. That is part of what I wrote in https://github.com/dotnet/machinelearning-samples/issues/143#issuecomment-443410619. In EF Core 3.0 this is getting easier to predict, because we reverted some of the previous decisions:

1. We will generate only one SQL query for each LINQ query, regardless of whether you have Include()
2. We don't do any identity resolution, which means that we don't keep weak references to the entities already returned.